File: sic-api/Model/Pageable.cs
namespace sic_api.Model;

public class Pageable
{
    public int PageNumber { get; set; } = 1;
    public int PageSize { get; set; } = 10;
    public long TotalElements { get; set; }
    public long TotalPages => (TotalElements + PageSize - 1) / PageSize;
    public PageableSort[] Sorts { get; set; } = [];
    public bool HasSort => Sorts.Length > 0;
}

public class PageableCombobox
{
    public string? Keyword { get; set; }
    public Guid? Value { get; set; }
    public int PageNumber { get; set; } = 1;
    public int PageSize { get; set; } = 10;
    public long TotalElements { get; set; }
    public long TotalPages => (TotalElements + PageSize - 1) / PageSize;
    public PageableSort[] Sorts { get; set; } = [];
    public bool HasSort => Sorts.Length > 0;
}

public class PageableSort
{
    public string Field { get; set; } = default!;
    public bool Descending { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/dto/Pageable.java
package com.softinter.sicapi.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pageable {
    private int pageNumber = 1;
    private int pageSize = 10;
    private long totalElements;
    private long totalPages;
    private List<PageableSort> sorts = new ArrayList<>();

    public boolean hasSort() {
        return sorts != null && !sorts.isEmpty();
    }

    public void calculateTotalPages() {
        this.totalPages = (totalElements + pageSize - 1) / pageSize;
    }
}
